src/views/Workspace/Sections/WorkspaceWelcome/WorkspaceWelcome.jsx

```diff
@@ -53,7 +53,7 @@ class WorkspaceWelcome extends React.Component {
     }
     
     goToLogin() {
-        history.push('/login');
+        history.push(`/login?from=${this.props.location.pathname}&id=${this.props.playground.id}`);
```

Comment:
The url already includes the playgroundid. I do not think it is necessary to define the additional id parameter

Reply:
Actually yes. If we always set a target URL for the login component we will use only it.